### ISSUE TITLE
[RFC] Business Req. proposal from issues

### ISSUE THREAD

**Issue by rubencabrera** [OCA/business-requirement#171]
The process suggested in this repo's modules, as I understand it, is collecting customer's needs in BRs and having the option to create tasks and projects from there. 

That's very useful when starting a project, but in software development it is quite common to receive BRs in the form of project issues, especially when it is a Support customer. 

I'm thinking about a module to cover this process:
+ An issue is opened.
+ The issue is evaluated and it happens to need development out of the current scope of the project. 
+ "Escalate" the issue to a BR (or any other object of this repo you suggest).
+ From the BR, continue with the usual process and then send quotes, create tasks or any other action of the BR workflow. 

Any suggestions or ideas to make this work for everyone?

**Comment by rubencabrera** [comment]
The  business_requirement_deliverable_crm module links opportunities and BRs, but assumes that it always starts from a lead. The case I am presenting is when the requirement comes directly to tech support and has to be transfered to business consultants or salesmen.

**Comment by dreispt** [comment]
I agree that the described use case is relevant and should be considered.

**Comment by rubencabrera** [comment]
Changed the title to narrow it down only to issues, as tasks can be created from BR when the proposal is approved and put to development. Please comment if anyone thinks of a case when this should be different.

Also gave it a bit of thought and it seems to me that linking the issue to the BR will be fine, as it fits perfectly (IMO) what the process of change proposal and approval involves.

**Comment by dreispt** [comment]
Preferring to link with Issues sounds good to me.
If we can avoid a may2many, I suggest having 1 Issue -> N BRs.
The Issue should have a button/action "Create BR".

**Comment by elicoidal** [comment]
We currently are thinking of refactoring the links between tasks, projects and BR via #115 
Basically we think that the BR should be linked directly to a project and not tasks. In this view, a new task or new issue associated to a project would be accounted directly for the given related BR.
(I invite you to comment on original RFC for this topic)

Back to your question: I think it is valid with following details on my side:
Multiple cases for Issue:
1. can be related to existing implementation/development BR as part of debugging/support
2. can be linked to a separated BR for support (same treatment as above though)
3. can be not linked to existing BR (new feature, change request)

1 and 2 could be covered directly adding the issue to the existing project.
3. Could be initially linked to existing project but after assessment might become a new BR.

In my opinion that could be a good moment to add a "Change Request of" link to BR (m2o) along with smart button to display all BR "change request " of existing on.

I like the idea of having a button in the Issue which would allow to create a new BR but it raises some difficulty:
* project from BR is created down the road, not at the beginning (so the button could not link the issue directly to any project right away)
* It might be difficult to change the project if validated timesheets are linked to the issue.

In short, it is possible to create the BR but seems unrealistic to reassign the issue to the newly created BR (annoying but not blocking).

**Comment by dreispt** [comment]
For me it is not natural to have to create a new Project for a Change Request.
I see a CR as an amendment/addition to an existing Budget/Project.
Sure BRs must be linked to Projects, but I don't see why you can't link to an existing project rather than being forced to create a new Project.
So the link from Issue <-> BR is only instrumental; the important link will be from that BR <-> Project (of the origin Issue).

**Comment by rubencabrera** [comment]
On our use case, the instrumental point of view fits perfectly.
Issues are managed by tech support, BRs by consultants, this button is just a way of telling them about a problem that needs analysis. The default project is the one assigned to the issue, usually automated by an alias from incoming emails. 

The "change request of" is a very interesting feature but maybe for another module.

**Comment by dreispt** [comment]
@rubencabrera This _is_ a CR implementation: CR is just a concept; it is implemented as an Issue with a linked BR.

**Comment by elicoidal** [comment]
> The default project is the one assigned to the issue, usually automated by an alias from incoming emails.

Speaking of which, you know that you can create alias for the BR to be created automatically?

@dreispt @rubencabrera OK with the requirement which is add a button in issue to automatically create a BR with following information:
* Name: name of the issue
* Project: project of the issue
* Responsible: assignee of the issue or current user

Do we need anything else in a wizard?

**Comment by dreispt** [comment]
> Do we need anything else in a wizard?

If it is, you can always add it later.

**Comment by elicoidal** [comment]
True!

**Comment by rubencabrera** [comment]
Already added some more default fields that seemed relevant. The user_id is used for multiple fields where usually another user will be assigned, but as an entry value it's good to have something.
